src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/RefreshTokenTest.java

```diff
@@ -0,0 +1,128 @@
+
+    @DisplayName("Refresh tokens when both access and refresh tokens are valid")
+    @Test
+    @WithMockUser
+    void refreshTokensWithCorrectTokens_thenReturnNewAccessTokenAndSetNewRefreshTokenInCookie() throws Exception {
+        User user = new User();
+
+        when(userService.validateToken(validAccessToken)).thenReturn(true);
+        when(userService.validateToken(validRefreshToken)).thenReturn(true);
+        when(userService.getUserFromToken(validAccessToken)).thenReturn(user);
+        when(userService.createAccessToken(user)).thenReturn(newAccessToken);
+        when(userService.createRefreshTokenCookie(user)).thenReturn(newRefreshTokenCookie);
+
+        mockMvc.perform(post("/api/auth/refresh")
+                        .cookie(new Cookie("refresh_token", validRefreshToken))
+                        .header("Authorization", "Bearer " + validAccessToken))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.accessToken").value(newAccessToken))
+                .andExpect(result -> {
+                    String setCookieHeader = result.getResponse().getHeader("Set-Cookie");
+                    assertTrue(setCookieHeader.contains("refresh_token=" + newRefreshToken));
+                });
+    }
+
+    @DisplayName("Attempt to refresh tokens without refresh token")
+    @Test
+    void refreshTokenWithoutRefreshToken_thenReturnUnauthorized() throws Exception {
+        mockMvc.perform(post("/api/auth/refresh")
+                        .header("Authorization", "Bearer " + validAccessToken))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @DisplayName("Attempt to refresh tokens with incorrect access token")
+    @Test
+    void refreshTokenWithIncorrectAccessToken_thenReturnUnauthorized() throws Exception {
+        when(userService.validateToken(invalidAccessToken)).thenReturn(false);
+
+        mockMvc.perform(post("/api/auth/refresh")
+                        .cookie(new Cookie("refresh_token", validRefreshToken))
+                        .header("Authorization", "Bearer " + invalidAccessToken))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @DisplayName("Attempt to refresh tokens with incorrect refresh token")
+    @Test
+    void refreshTokenWithIncorrectRefreshToken_thenReturnUnauthorized() throws Exception {
+        when(userService.validateToken(validAccessToken)).thenReturn(true);
+        when(userService.validateToken(invalidRefreshToken)).thenReturn(false);
+
+        mockMvc.perform(post("/api/auth/refresh")
+                        .cookie(new Cookie("refresh_token", invalidRefreshToken))
+                        .header("Authorization", "Bearer " + validAccessToken))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @DisplayName("Attempt to refresh tokens with expired access token")
+    @Test
+    void refreshTokenWithExpiredAccessToken_thenReturnUnauthorized() throws Exception {
```

Comment:
as i mentioned earlier user should be able to refresh expired access token in particular